spec/integration/network/http_pool_spec.rb

```diff
@@ -170,6 +170,7 @@ def require_client_certs(ctx)
 
     it "warns when client has an incomplete client cert chain" do
       expect(Puppet).to receive(:warning).with("The issuer 'CN=Test CA Agent Subauthority' of certificate 'CN=pluto' cannot be found locally")
+      expect(Puppet).to receive(:warning).with(/Puppet::Network::HTTP::Connection is deprecated/)
```

Comment:
I'd expect the `HttpPool.connection` method to use the `ConnectionAdapter` and not trigger this deprecation warning. You might need to rebase on `upstream/master` to pick up the change to `HttpPool.http_client_class` to the adapter class?

Reply:
I don't think I need this line, but I am up to date with master, and there are still test failures I wasn't expecting to hit


```
Failures:

  1) Puppet::Network::HttpPool when calling HttpPool.connection method warns when client has an incomplete client cert chain
     Failure/Error: warning("%{message}\n   %{call_trace}" % { message: message, call_trace: call_trace })
       Puppet received :warning with unexpected arguments
         expected: ("The issuer 'CN=Test CA Agent Subauthority' of certificate 'CN=pluto' cannot be found locally")
              got: ("Puppet::Network::HTTP::Connection is deprecated. Please use Puppet::Network::HTTP::ConnectionAdapter... (location: /Users/melissa/workdir/puppet/lib/puppet/network/http/connection.rb:52:in `initialize')")
       Diff:
       @@ -1,2 +1,3 @@
       -The issuer 'CN=Test CA Agent Subauthority' of certificate 'CN=pluto' cannot be found locally
       +Puppet::Network::HTTP::Connection is deprecated. Please use Puppet::Network::HTTP::ConnectionAdapter instead.
       +   (location: /Users/melissa/workdir/puppet/lib/puppet/network/http/connection.rb:52:in `initialize')
     # ./lib/puppet/util/logging.rb:298:in `issue_deprecation_warning'
     # ./lib/puppet/util/logging.rb:143:in `deprecation_warning'
     # ./lib/puppet/network/http/connection.rb:52:in `initialize'
     # ./lib/puppet/network/http/connection_adapter.rb:3:in `initialize'
     # ./lib/puppet/network/http_pool.rb:85:in `new'
     # ./lib/puppet/network/http_pool.rb:85:in `connection'
     # ./spec/integration/network/http_pool_spec.rb:193:in `block (4 levels) in <top (required)>'
     # ./spec/lib/puppet_spec/https.rb:81:in `block in start_server'
     # ./spec/lib/puppet_spec/https.rb:41:in `pipe'
     # ./spec/lib/puppet_spec/https.rb:41:in `start_server'
     # ./spec/integration/network/http_pool_spec.rb:192:in `block (3 levels) in <top (required)>'
     # ./spec/spec_helper.rb:180:in `block (2 levels) in <top (required)>'
     # util/rspec_runner:44:in `run'
     # util/rspec_runner:59:in `<main>'
  2) Puppet::Network::HttpPool when managing http instances when calling 'connection' warns if an ssl_context is used for an http connection
     Failure/Error: warning("%{message}\n   %{call_trace}" % { message: message, call_trace: call_trace })
       Puppet received :warning with unexpected arguments
         expected: ("An ssl_context is unnecessary when connecting to 'http://me:8140' and will be ignored")
              got: ("Puppet::Network::HTTP::Connection is deprecated. Please use Puppet::Network::HTTP::ConnectionAdapter... (location: /Users/melissa/workdir/puppet/lib/puppet/network/http/connection.rb:52:in `initialize')")
       Diff:
       @@ -1,2 +1,3 @@
       -An ssl_context is unnecessary when connecting to 'http://me:8140' and will be ignored
       +Puppet::Network::HTTP::Connection is deprecated. Please use Puppet::Network::HTTP::ConnectionAdapter instead.
       +   (location: /Users/melissa/workdir/puppet/lib/puppet/network/http/connection.rb:52:in `initialize')
     # ./lib/puppet/util/logging.rb:298:in `issue_deprecation_warning'
     # ./lib/puppet/util/logging.rb:143:in `deprecation_warning'
     # ./lib/puppet/network/http/connection.rb:52:in `initialize'
     # ./lib/puppet/network/http/connection_adapter.rb:3:in `initialize'
     # ./lib/puppet/network/http_pool.rb:92:in `new'
     # ./lib/puppet/network/http_pool.rb:92:in `connection'
     # ./spec/unit/network/http_pool_spec.rb:104:in `block (4 levels) in <top (required)>'
     # ./spec/spec_helper.rb:180:in `block (2 levels) in <top (required)>'
     # util/rspec_runner:44:in `run'
     # util/rspec_runner:59:in `<main>'
Finished in 24.27 seconds (files took 2.07 seconds to load)

Finished in 1 minute 25.82 seconds
25778 examples, 2 failures, 39 pending
```

Reply:
Also, both tests are calling `Puppet::Network::HttpPool.connection` and then checking warnings